doc/reference/autoinstall-reference.rst

```diff
@@ -18,25 +15,13 @@ Here is an example of a minimal autoinstall configuration:
        ...
 
 
-At the top level is the ``autoinstall`` keyword, which contains a version section
-and an (incomplete) identity section which are explained in more detail below.
-Any other key at the level of ``autoinstall``, will result in an autoinstall
-validation error at runtime.
-
-.. warning::
-
-   This behaviour was first introduced during 24.04 (Noble). On any ISOs built
-   before this, you will need to refresh the installer to see this behaviour.
-   Please the note below about the old format.
+At the top level is the ``autoinstall`` keyword. It contains a ``version`` section and an (incomplete) ``identity`` section, which are explained in more detail below. Any other key at the level of ``autoinstall`` results in an autoinstall validation error at run time.
 
 .. note::
 
-  Technically, in all but one case the top level ``autoinstall`` keyword is
-  strictly unnecessary. This keyword is only necessary when serving autoinstall
-  via cloud-config. For backwards compatibility this format is still supported
-  for non-cloud-config based delivery methods; however, it is
-  **highly recommended** to use the format with a top-level ``autoinstall``
-  keyword as mistakes in this formatting are a common source of confusion.
+    This behaviour was first introduced during 24.04 (Noble). On any ISOs built before 24.04, you need to refresh the installer to see this behaviour.
```

Comment:
I'm good to merge this PR but I still want wording tweaks on this note, I'll follow-up with a PR for that.